pandas/tests/api/test_api.py

```diff
@@ -221,6 +222,18 @@ def test_api(self):
         )
 
 
+def test_datetime():
+    from datetime import datetime
+    import pytest
+
+    msg = "The pandas.datetime module is deprecated " \
+          "and will be removed from pandas in a future version. " \
+          "Import datetime directly instead."
+
+    with pytest.warns(FutureWarning, match=msg):
```

Comment:
See the contributing guide - we have our own context manager for this: (this is whats causing code_checks to fail)

https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#testing-warnings